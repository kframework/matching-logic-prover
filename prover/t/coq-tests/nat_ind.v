Inductive nat : Type :=  Z : nat | S : forall (x : nat), nat .

// symbol nat() : Term
// symbol Z() : Term
// symbol S() : Term

// axiom \type(Z(), nat())

Definition nat_ind := 
fun (P : (forall (n : nat), Prop)) (f : P 0) (f0 : forall (n : nat), (forall (x : P n), P (S n))) =>
fix F (n : nat) := match n with
                           Z => f
                         | S n0 => f0 n0 (F n0)
                   end
(*
hasType (forall (P : (forall (x : nat), Prop)),
        (forall (y : P 0), (forall (l : (forall (n : nat), (forall (y : P n), P (S n)))), (forall (n : nat), P n))))
*)
.

// forall P : nat -> Prop, P 0 -> (forall n : nat, P n -> P (S n)) -> forall n : nat, P n
// tr(fun ... ) \mu { F {{ Term }} } tr( fun n => match ) ... )
