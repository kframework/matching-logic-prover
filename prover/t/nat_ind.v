Definition nat_ind := 
fun (P : nat -> Prop) (f : P 0) (f0 : forall (n : nat), P n -> P (S n)) =>
fix F (n : nat) := match n with
                           Z => f
                         | S n0 => f0 n0 (F n0)
                         end .

// tr(fun ... ) \mu { F {{ Term }} } tr( fun n => match ) ... )