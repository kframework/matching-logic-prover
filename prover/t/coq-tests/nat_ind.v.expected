<prover>
  <k>
    .CoqSentences
  </k>
  <exit-code>
    1
  </exit-code>
  <goals>
    .GoalCellSet
  </goals>
  <declarations>
    <declaration>
      axiom \equals ( nat_ind , \lambda { P { Term } , f { Term } , f0 { Term } , .Patterns } \mu { F {{ Term }} , .Patterns } \lambda { n { Term } , .Patterns } \or ( \exists { .Patterns } \and ( \equals ( n , Z ) , f , .Patterns ) , \exists { n0 , .Patterns } \and ( \equals ( n , S ( n0 , .Patterns ) ) , f0 ( n0 , F , n0 , .Patterns ) , .Patterns ) , .Patterns ) )
    </declaration> <declaration>
      axiom \type ( S ( .Patterns ) , \forall { x { Term } , .Patterns } nat )
    </declaration> <declaration>
      axiom \type ( Z ( .Patterns ) , nat )
    </declaration> <declaration>
      sort Term
    </declaration> <declaration>
      symbol S ( .Sorts ) : Term
    </declaration> <declaration>
      symbol Z ( .Sorts ) : Term
    </declaration> <declaration>
      symbol nat ( .Sorts ) : Term
    </declaration> <declaration>
      symbol nat_ind ( .Sorts ) : Term
    </declaration>
  </declarations>
</prover>
